Spring Context not loading with lazy-init from version 14.0.1

## Expected Behavior

Upgrade to version `14.0.1` is a patch version and does not include breaking changes. 

## Current Behavior

We are running spring boot integration tests with `spring.main.lazy-initialization: true`
Context cannot be loaded anymore:
```
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'com.github.kagkarlsson.scheduler.boot.autoconfigure.DbSchedulerAutoConfiguration': Failed to instantiate [com.github.kagkarlsson.scheduler.boot.autoconfigure.DbSchedulerAutoConfiguration$$SpringCGLIB$$0]: No default constructor found
```

The root cause is this change, to not eagerly load the DbSchedulerStarter:
https://github.com/kagkarlsson/db-scheduler/pull/492/files

--------
# For bug reports

## Context
- DB-Scheduler Version : 14.0.2
- Java Version : 21
- Spring Boot (check for Yes) : [X]
- Database and Version : Postgres 16

--- a/db-scheduler-boot-starter/src/main/java/com/github/kagkarlsson/scheduler/boot/autoconfigure/DbSchedulerAutoConfiguration.java
+++ b/db-scheduler-boot-starter/src/main/java/com/github/kagkarlsson/scheduler/boot/autoconfigure/DbSchedulerAutoConfiguration.java
@@ -194,7 +194,7 @@ public class DbSchedulerAutoConfiguration {
   }
 
   @Bean
-  public LazyInitializationExcludeFilter eagerDbSchedulerStarter() {
+  static LazyInitializationExcludeFilter eagerDbSchedulerStarter() {
     return LazyInitializationExcludeFilter.forBeanTypes(DbSchedulerStarter.class);
   }
 

